### ISSUE TITLE
[Feature request]: Use a more recent version for modals

### ISSUE THREAD

**Issue by idotj** [Jisan-mia/dom-projects#40]
### Type of feature

🧑‍💻 Code Refactor

### Is your feature request related to a problem? Please describe.

Modal example works as expected:
https://www.jisan.io/dom-projects/projects/modal/

But fails if we check WCAG standards (for example: impossible to close the modal with Esc key).

### Describe the solution you'd like.

Now that we are in 2024, I will suggest to use Dialog tag that works in all browsers and simplifies a lot the way to develop a modal.

If you prefer to keep the current one as "legacy" version then will be better to create a new one.

### Describe alternatives you've considered.

Here is the MDN documentation with some examples:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog

### Additional context

If this seems plausible, I can give a hand and develop an example.

### Record

- [X] I have checked the existing [issues](https://github.com/Jisan-mia/dom-projects/issues)
- [X] I have read the [Contributing Guidelines](https://github.com/Jisan-mia/dom-projects/blob/main/CONTRIBUTING.md)
- [X] I want to work on this issue

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Jisan-mia** [comment]
Hi @idotj , 
Thanks a lot for this improvement feedback. the new HTML Dialog element looks promising. Especially the keyboard ESC key functionality is very much needed for better accessibility like you mentioned. 
Would you like to work on this issue? so that I can assign this issue to you

**Comment by idotj** [comment]
Sure, feel free to assign the issue to me.
Just a question, you want to keep the current one (so I create a new project/folder) or better to update the code in the current [modal](https://github.com/Jisan-mia/dom-projects/tree/main/projects/modal) project?

**Comment by Jisan-mia** [comment]
We can keep the existing modal as it is, and create a new project using the native Dialog element, by this we can include both the Dialog element implementation and javascript implementation

What do you think about this? @idotj

**Comment by Jisan-mia** [comment]
@sadiul-hakim bhaiya, we can keep this issue open until there is a pr raised

**Comment by idotj** [comment]
Hi @Jisan-mia 
I raised the PR with the new Modal. Have a look and let me know if you have any questions.

FYI, I would like to share some feedback about some of the projects I saw in your repository:

- Try to put your script tag in the HTML file at the very bottom of your document (before closing the body tag)
- Use HTML tags to structure your document (main, header, footer, aside, etc...)
- Please don't use in CSS files the outline:none for buttons with :focus (it's discouraged by WCAG standards because blocks users to see the focus when navigating with the TAB key)

**Comment by Jisan-mia** [comment]
Hi @idotj 
Those are some excellent points, we should follow in this project. Thanks a lot for this feedback. I'll cross-check every project listed with your suggestions.

Also, you're asking if you should put your WCAG-supported Modal next to the existing modal. Let's do this for future projects also so that people can easily view them one after another like you mentioned.

And for id, we can just increment by one from the last project id...

Can you please update your pr like this.. just add your project next to the existing modal in the data.json file

**Comment by idotj** [comment]
Done
